diff --git a/arch/arm/boot/dts/bcm2835-rpi-b.dts b/arch/arm/boot/dts/bcm2835-rpi-b.dts
index 58a0d60..dbcdf34 100644
--- a/arch/arm/boot/dts/bcm2835-rpi-b.dts
+++ b/arch/arm/boot/dts/bcm2835-rpi-b.dts
@@ -14,7 +14,7 @@
 
 		act {
 			label = "ACT";
-			gpios = <&gpio 16 1>;
+			gpios = <&gpio 47 1>;
 			default-state = "keep";
 			linux,default-trigger = "heartbeat";
 		};
@@ -23,7 +23,7 @@
 
 &gpio {
 	pinctrl-names = "default";
-	pinctrl-0 = <&gpioout &alt0 &alt2 &alt3>;
+	pinctrl-0 = <&gpioout &alt0 &alt3>;
 
 	gpioout: gpioout {
 		brcm,pins = <6>;
@@ -39,12 +39,6 @@
 		brcm,pins = <48 49 50 51 52 53>;
 		brcm,function = <7>; /* alt3 */
 	};
-
-	/* I2S interface */
-	alt2: alt2 {
-		brcm,pins = <28 29 30 31>;
-		brcm,function = <6>; /* alt2 */
-	};
 };
 
 &i2c0 {
